### PR TITLE
implemented isMiaHeaderInjected option

### PR DESCRIPTION
**Pull request by dibericky**
#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [x] the documentation is updated or intregrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

File: tests/serviceBuilder.test.js [hunk 1]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
     nock.enableNetConnect()
   })
 
+  test.test('getHeader', innerTest => {
+    const HEADER_MIA_KEY = 'miaheader'
+    const HEADER_MIA = { [HEADER_MIA_KEY]: 'foo' }
+
+    innerTest.test('injects Mia Header if isMiaHeaderInjected option is missing', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
+        .replyContentLength()
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: this should be useless

File: tests/serviceBuilder.test.js [hunk 2]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
     nock.enableNetConnect()
   })
 
+  test.test('getHeader', innerTest => {
+    const HEADER_MIA_KEY = 'miaheader'
+    const HEADER_MIA = { [HEADER_MIA_KEY]: 'foo' }
+
+    innerTest.test('injects Mia Header if isMiaHeaderInjected option is missing', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: you can specify the headers of the request. See https://github.com/nock/nock#specifying-request-headers

File: tests/serviceBuilder.test.js [hunk 3]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
...
+        .replyContentLength()
+        .get('/foo')
+        .reply(function response() {
+          assert.equal(this.req.headers[HEADER_MIA_KEY], HEADER_MIA[HEADER_MIA_KEY])
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: see comment line 43

File: tests/serviceBuilder.test.js [hunk 4]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
...
+      const response = await service.get('/foo', {}, { returnAs: 'JSON' })
+
+      assert.equal(response.statusCode, 200)
+
+      myServiceNameScope.done()
+      assert.end()
+    })
+
+    innerTest.test('injects Mia Header if isMiaHeaderInjected option is true', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: see above

File: tests/serviceBuilder.test.js [hunk 5]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
...
+      const response = await service.get('/foo', {}, { returnAs: 'JSON', isMiaHeaderInjected: true })
+
+      assert.equal(response.statusCode, 200)
+
+      myServiceNameScope.done()
+      assert.end()
+    })
+
+    innerTest.test('does not inject Mia header if isMiaHeaderInjected option is false', async assert => {
+      const myServiceNameScope = nock('http://my-service-name')
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: you can specify the bad headers headers of the request. See https://github.com/nock/nock#specifying-request-headers

File: lib/serviceBuilder.js [hunk 6]
```diff
@@ -164,7 +164,12 @@ function makeCall(hostname, path, method, querystring, body, options = {}) {
 }
 
 function getHeader(options, miaHeaders, baseOptions) {
-  return Object.assign({}, miaHeaders || {}, baseOptions.headers || {}, options.headers || {})
+  const isMiaHeaderInjected = options.isMiaHeaderInjected !== undefined ? options.isMiaHeaderInjected : true
```
Comment (Lucascanna) [inline lib/serviceBuilder.js]: why do you need this line?

File: tests/serviceBuilder.test.js [hunk 7]
```diff
@@ -35,6 +35,68 @@ tap.test('serviceBuilder', test => {
     nock.enableNetConnect()
   })
 
+  test.test('getHeader', innerTest => {
```
Comment (Lucascanna) [inline tests/serviceBuilder.test.js]: test title is a bit misleading. what about "forwarding of mia headers"?

File: lib/serviceBuilder.js (hunk 6 above)
Comment (dibericky) [reply lib/serviceBuilder.js]: Because I have to manage the default case:` isMiaHeaderInjected === true` when `options.isMiaHeaderInjected` is missing

Comment (Lucascanna) [reply lib/serviceBuilder.js]: `const isMiaHeaderInjected = options.isMiaHeaderInjected === undefined ? true : options.isMiaHeaderInjected` is more readable

**Comment by Lucascanna** [comment]
Manca l'aggiornamento del typescript

**Comment by davidebianchi** [comment]
LGTM

**Pull request merged**